plugins/generator-1.17.1/forge-1.17.1/templates/elementinits/potions.java.ftl

```diff
@@ -0,0 +1,62 @@
+<#--
+ # MCreator (https://mcreator.net/)
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2021, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+
+/*
+ *    MCreator note: This file will be REGENERATED on each build.
+ */
+
+package ${package}.init;
+
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD) public class ${JavaModName}Potions {
+
+    private static final List<Potion> REGISTRY = new ArrayList();
+
+    <#list potions as potion>
+    public static Potion ${potion.getModElement().getRegistryNameUpper()} = register(new Potion(
+        <#list potion.effects as effect>
+        new MobEffectInstance(${effect.effect?replace("Effect.effect", ".potion")?replace("Potion.potion", "PotionEffect.potion")}, ${effect.duration}, ${effect.amplifier}, ${effect.ambient}, ${effect.showParticles})<#if effect?has_next>,</#if>

```

Comment:
This part of the code (the replace part) will probably be different once the effect in 1.17.1 generator is added, so it may be left out for now, except if it has a purpose for built-in effects too.

Or maybe even better, we should wait for the effects to be ported and then add this one, so everything is tested at once.

Reply:
@AlsoSomeoneElse this can now be adapted as effect mod element was merged